cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java

```diff
@@ -195,13 +198,14 @@ public Integer call() throws Exception {
     }, timeoutSeconds, TimeUnit.SECONDS, 50, TimeUnit.MILLISECONDS);
   }
 
-
   private boolean isProgramRunning(ProgramRuntimeService runtimeService, final Id.Program program) {
-    return runtimeService.checkAnyRunning(new Predicate<Id.Program>() {
+    Iterable<Id.Program> runningPrograms = Iterables.filter(runtimeService.getAllRunning(ProgramType.values()),
+                                                            new Predicate<Id.Program>() {
       @Override
-      public boolean apply(Id.Program programId) {
-        return programId.equals(program);
+      public boolean apply(Id.Program program) {
+        return program.getNamespaceId().equals(program);
```

Comment:
I don't understand. We were comparing programId(s) before. Why did that logic change to comparing programId and namespaceId?